sway-core/src/semantic_analysis/namespace/trait_map.rs

```diff
@@ -71,10 +71,17 @@ impl OrdWithEngines for TraitKey {
 /// Map of name to [TyImplItem](ty::TyImplItem)
 type TraitItems = im::HashMap<String, TyImplItem>;
 
+#[derive(Clone, Debug)]
+struct TraitValue {
+    trait_items: TraitItems,
+    // The span of the entire impl block.
```

Comment:
```suggestion
    /// The span of the entire impl block.
```

Reply:
We already have a few functions in the server that return `impl Iterator<Item= Span>` so I think it's a good idea to keep this practice going.

Reply:
For example our `tokens_for_file` function...

```rust
pub fn tokens_for_file<'s>(
    &'s self,
    uri: &'s Url,
) -> impl 's + Iterator<Item = (Ident, Token)> {
    self.iter().flat_map(|(ident, token)| {
        ident.span().path().and_then(|path| {
            if path.to_str() == Some(uri.path()) {
                Some((ident.clone(), token.clone()))
            } else {
                None
            }
        })
    })
}
```

Reply:
I agree in principle, but I'm not sure if this is necessary when dealing with such small collections of data. Thoughts?

Reply:
I agree it's probably not necessary but it would be nice to go this direction from a unified API perspective. Just an idea I don't have any stronger opinions on this other than that.

Reply:
I actually need the results as a vector anyway, so making this function return an iterator would add some overhead (converting to iter and then back to vec). In the compiler, the existing functions that are similar to this are returning vector, so I think it makes more sense to stick with that in this case.